### ISSUE TITLE
autoFixOnSave requires 2 saves when used in conjunction with tslint.run = onSave

### ISSUE THREAD

**Issue by brianpaulcarpenter** [Microsoft/vscode-tslint#303]
- tslint Version: 5.7.0
- typescript Version: 2.6.2

The default setting for tslint.run is onType, but I find the red squigglies appearing as you type a little distracting and would prefer them to come after I've finished writing the entire line of code.  There is an onSave option which does that (specifically it lints on save), but when used in combination with autoFixOnSave it requires 2 saves -- the first save will find the lint violations and the 2nd save will fix them.

It does not appear to be a race since it's happening 100% of the time.  It would be nice if autoFixOnSave would first wait for the lint violations to be discovered if the user has linting triggered onSave.

**Comment by cancerberoSgx** [comment]
@egamma maybe is not the same issue. In my case I installed this extension from marketplace and I needed to execute "tslint: fix all auto fixable problems" several times in order to do the complete indentation. That's the issue I'm talking about here. 

Then I `npm install tslint` and run it from command line` tslint --fix`  and I noticed that it worked as expected (indenting my files totally and only in one pass)

hope is clear. 

BTW: at some point I observed the same behavior with eslint vscode extension but no more. So my guess the issue is in vscode itself and was fixed or something like that.
